docs/pages/api-reference/liveblocks-client.mdx

```diff
@@ -891,18 +891,142 @@ Note that the function should be called like this `defineAiTool()({ ... })`
   </PropertiesListItem>
   <PropertiesListItem name="execute" type="function">
     Async function that performs the tool's action. Receives validated arguments
-    and execution context, returns structured data.
+    and execution context, returns structured data. See [implementing the tool
+    call via `execute`](#implement-via-execute).
   </PropertiesListItem>
   <PropertiesListItem name="render" type="function">
     React component function that renders the tool's UI during different
-    execution stages.
+    execution stages. See [tool call rendering
+    stages](#tool-call-rendering-stages).
   </PropertiesListItem>
 </PropertiesList>
 
 Tools can be registered globally with
 [`RegisterAiTool`](/docs/api-reference/liveblocks-react#RegisterAiTool) or
 passed directly to [`AiChat`](/docs/api-reference/liveblocks-react-ui#AiChat).
 
+### Tool call rendering stages [#tool-call-rendering-stages]
+
+Rendering a tool call can be done before the tool call is executed, which allows
+you to display a UI during its entire lifecycle. The tool call stages are:
+
+- `receiving` The tool call is being received. This stage is reserved for future
+  use, to enable showing a UI while the tool call `args` object is being
+  streamed in, but before the tool call can actually be made.
+- `executing` The tool call is currently executing, or is ready to be. In this
+  stage, the `args` are fully known, but the result of the tool call is not
+  known yet.
+- `executed` The tool call has been executed, and the result is known. This
+  happens after your `execute` function was run, or after you called `respond()`
+  inside `render`. In this stage, the `result` object will be available.
+
+The render component will automatically re-render when its stage changes.
```

Comment:
`defineAiTool` isn't React-specific but since this page is for `@liveblocks/client` I think we would could specify that `render` is React-specific.